configs/components/rubygem-nokogiri.rb

```diff
@@ -5,14 +5,18 @@
     # including its dependencies (mini_portile2), installing it on the target, and
     # installing the nokogiri gem using our gem command. The added files from
     # :gem_home were then tar'ed up and packed into this container tarball.
-    pkg.url "http://buildsources.delivery.puppetlabs.net/nokogiri-precompiled-huaweios-#{pkg.get_version}.tar.gz"
-    pkg.md5sum "8a783bdda33cf5f10afe84cf6fd8f28c"
+    pkg.url "http://buildsources.delivery.puppetlabs.net/nokogiri-precompiled-huaweios-#{pkg.get_version}-for-ruby-#{settings[:ruby_version]}.tar.gz"
+    if settings[:ruby_version] == "2.1.9"
```

Comment:
It kind of seems to me like a cleaner way to do this would to have the huaweios precompiled nokogiri be it's own separate source with a separate component file, and then select it based on platform/ruby version.. i don't know if that's possible though..

I.e., 
`configs/components/rubygem-nokogiri-huaweio-2.3.1.rb`
`configs/components/rubygem-nokogiri-huaweio-2.1.9.rb`
`configs/components/rubygem-nokogiri.rb`

and then select the right one based on what platform/ruby version we're building?